scripts/microbe_stage/species_system.lua

```diff
@@ -0,0 +1,211 @@
+    --end
+    organelleData = storage:get("organelleData", nil)
+    self.organelles = {}
+    if organelleData ~= nil then
+        i = 1
+        while organelleData:contains(""..i) do
+            organelle = {}
+            orgData = organelleData:get(""..i, nil)
+            if orgData ~= nil then
+                organelle.name = orgData:get("name", "")
+                organelle.q = orgData:get("q", 0)
+                organelle.r = orgData:get("r", 0)
+            end
+            self.organelles[i] = organelle
+            i = i + 1
+        end
+    end
+end
+
+--todo
+function SpeciesComponent:storage()
+    local storage = Component.storage(self)
+    storage:set("name", self.name)
+    storage:set("populationBonusFactor", self.populationBonusFactor)
+    storage:set("populationPenaltyFactor", self.populationPenaltyFactor)
+    storage:set("deathsPerTime", self.deathsPerTime)
+    storage:set("birthsPerTime", self.birthsPerTime)
+    storage:set("currentPopulation", self.currentPopulation)
+    compoundPriorities = StorageContainer()
+    for k,v in pairs(self.compoundPriorities) do
+        compoundPriorities:set(k,v)
+    end
+    storage:set("compoundPriorities", compoundPriorities)
+    organelles = StorageContainer()
+    for i, org in ipairs(self.organelles) do
+        orgData = StorageContainer()
+        orgData:set("name", org.name)
+        orgData:set("q", org.q)
+        orgData:set("r", org.r)
+        organelles:set(""..i, orgData)
+    end
+    storage:set("organelleData", organelles)
+    return storage
+end
+
+function SpeciesComponent:mutate(aiControlled, population)
+    --[[
+    SpeciesComponent:mutate should call the correct evolution-handler
+    (Ie, trigger an entry into microbe editor for player; and auto-evo 
+    for AI), and then create and add the new species to the system.
+    - uses the population to access information about 
+    
+    For player mutation, this would be the entry point into the editor
+
+    - note, this architecture may be reconsidered
+
+    ]]
+end
+
+
+-- Given a newly-created microbe, this sets the organelles and all other species-specific microbe data
+--  like agent codes, for example.
+function SpeciesComponent:template(microbe)
+    microbe.microbe.speciesName = self.name
+    -- give it organelles
+    for i, orgdata in pairs(self.organelles) do
+        organelle = OrganelleFactory.makeOrganelle(orgdata)
+        microbe:addOrganelle(orgdata.q, orgdata.r, organelle)
+    end
+
+    for compoundID, amount in pairs(self.avgCompoundAmounts) do
+        if amount ~= 0 then
+            microbe:storeCompound(compoundID, amount, false)
+        end
+    end
+    for compoundID, priority in pairs(self.compoundPriorities) do
+        if priority ~= 0 then
+            microbe:setDefaultCompoundPriority(compoundID, priority)
+        end
+    end
+    -- complimentary serving of atp
+    --newMicrobe:storeCompound(CompoundRegistry.getCompoundId("atp"), 10)
```

Comment:
Priority should be on the microbe with the current system, as it constantly updates priorities based on current compounds.
Having some defaults defined in the species that then gets puhed to microbes on creation would probably be a good idea tho